### PR TITLE
INT-319: Introduce Drag and Drop to Embed + ConnectorURL Fix

### PR DESCRIPTION
**Pull request by kirisanth-g**
### What
* Introduce `dragAndDrop` option to embed
* If turned on Button & Intros will be draggable
* Button/Intro will not active during drag
* Added cluster variable to connector's url
### Test
* Load embed and drag it around
* Open & Close will no longer make the button disappear
---
- [ ] PR title follows the format \<Jira Card\>: \<Change description\>
- [ ] PR description present and accurate
- [ ] PR is linked to a Jira card has `impact` field filled in
- [ ] Unit tests cover code changes
- [ ] Changes have gone through design review if required
- [ ] Deployment plan and/or bridge code description is included in PR if required
- [ ] Changes are CATO/WCAG compliant and support IE9+
- [ ] A [Release Note](https://www.notion.so/adasupport/Creating-Release-Notes-36906dfa8a2b4f10a31dc23b8d46681a) has been drafted and published after merging to master

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,8 +52,8 @@
     "@types/classnames": "^2.2.7",
     "classnames": "^2.2.6",
     "es6-promise-promise": "^1.0.0",
-    "preact": "^8.2.5",
-    "preact-compat": "^3.17.0",
+    "preact": "^8.4.2",
```
Comment (NicHaley) [inline package.json]: Didn't notice we had updated the preact version earlier. What for exactly? We should do a little extra testing to make sure all good

Comment (kirisanth-g) [reply package.json]: This was apart of the `v1.0.0` release for `<style>`, GitHub seems to be showing changes dating back from January for some reason 😕

Comment (kirisanth-g) [reply package.json]: I'm wondering if merging back into beta from master is causing this weird error

**Comment by brandonmowat** [comment]
@kirisanth-g Can we fill in the PR description?

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -1 +0,0 @@
-* @NicHaley @rinoc @brandonmowat @sevarubbo @kirisanth-g
```
Comment (sevarubbo) [inline .github/CODEOWNERS]: Why?

File: src/components/App/Draggable/index.tsx [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+    }
+  }
+
+  delayedSetter() {
+    setTimeout(
+      (() => {
+        this.setState({ beingDragged: false });
+        document.removeEventListener("click", this.handleOnClick, true);
+      })
+      .bind(this),
```
Comment (sevarubbo) [inline src/components/App/Draggable/index.tsx]: No need for .bind here

File: src/components/App/Draggable/index.tsx [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+  handleOnClick(e) {
+    e.stopPropagation();
+    e.preventDefault();
+  }
+
+  onMouseDown(e: MouseEvent) {
+    // only left mouse button
+    if (e.button !== 0) return;
+    const pos = this.myRef.getBoundingClientRect();
```
Comment (sevarubbo) [inline src/components/App/Draggable/index.tsx]: `pos` is not used

File: src/components/App/Draggable/index.tsx [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+      position: "fixed",
+      right: `${pos.x}px`,
+      bottom: `${pos.y}px`
+    };
+
+    // Because Preact children is defined so oddly -_-
+    const numChildren = Object.keys(children).length;
+    const newChildren = [];
+    let i: number;
+    for (i = 0; i < numChildren; i = i + 1) {
```
Comment (sevarubbo) [inline src/components/App/Draggable/index.tsx]: You can just do `Object.keys(children).forEach`. But even better to do `Object.keys(children).filter(...).map(...)`

File: src/services/browsers.ts [hunk 6]
```diff
@@ -5,3 +5,10 @@ export function isIE9OrBelow(): boolean {
   return /MSIE\s/
     .test(navigator.userAgent) && parseFloat(navigator.appVersion.split("MSIE")[1]) < 10;
 }
+
+/**
+ * Detect if device is iOS
+ */
+export function isIOS(): boolean {
+  return /iPad|iPhone|iPod/.test(navigator.userAgent);
+}
```
Comment (sevarubbo) [inline src/services/browsers.ts]: Don't forget an empty line at the end. You can configure your IDE to do this automatically

File: preact-beta.config.js [hunk 7]
```diff
@@ -1,17 +1,35 @@
...
-
-  // This is to fix the issue where the compiled CSS classnames were given a localIdentName of
-  // [local]__[hash], but this did not match the component class names
-  const { loader: cssLoader } = helpers.getLoadersByName(config, "css-loader")[0];
-  cssLoader.options.localIdentName = "[local]";
+  const cssLoaders = helpers.getLoaders(config).filter(loader => {
+    return loader.rule.test.toString() === `/\\.(css|less|s[ac]ss|styl)$/`.toString()
+  });
 
+  cssLoaders.forEach(loader => {
```
Comment (sevarubbo) [inline preact-beta.config.js]: It's weird, this changes are already in master...
https://github.com/AdaSupport/embed/blob/master/preact-beta.config.js

If there's a github bug, maybe it makes sense to close and reopen the PR?

File: .github/CODEOWNERS (hunk 2 above)
Comment (sevarubbo) [reply .github/CODEOWNERS]: Ok, i got it

File: package.json (hunk 1 above)
Comment (NicHaley) [reply package.json]: hmm prob because master wasn't merged _back_ into beta after we merged master. Can you try merging master into beta

**Pull request merged**